databases/week3/Meal_application.sql

```diff
@@ -0,0 +1,372 @@
+
+SELECT * FROM Review;
+
+-- Add a new Review
+INSERT INTO Review (Title, `Description`, Meal_id, Stars, Created_date) 
+VALUES (
+'Excellent', 
+'incredible food',
+'2', 
+4, 
+'2020-02-14'
+);
+
+-- Get a Review with any id...
+
+SELECT * FROM Review
+WHERE id = 1; 
+
+-- Update a Review with any id...
+
+UPDATE Review 
+SET Title = 'Fatoush', `Description` = 'great salad ever' , Meal_id = 4, Stars = 5, Created_date = '2020-02-012'
+WHERE Id = 2; 
+
+-- Delete a Review with any id
+DELETE FROM Review 
+WHERE Id = 3; 
+
+
+-------------------------------------------
+
+-- Additional queries...
+
+-- add a couple of different meals, reservations and reviews with different attributes.
+
+create database meal_application02;
+use meal_application02;
+
+SET NAMES utf8mb4;
+
+
+
+CREATE TABLE `Meal` (
+ 
+`id` int (10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  
+`title` varchar(255) NOT NULL,
+  
+`description` text NOT NULL,
+  
+`location` varchar(255) NOT NULL,
+  
+`when` datetime timestamp,
+  
+`max_reservations` int (10) unsigned NOT NULL,
+  
+`price` decimal (4) NOT NULL,
+  
+`created_date` date
+ NOT NULL
+);
+
+
+
+CREATE TABLE `Reservation` (
+  
+`id` int (10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  
+`number_of_guests` int (10) unsigned NOT NULL,
+  
+`meal_id` int (10) unsigned NOT NULL,
+  
+`created_date` date
+ NOT NULL);
+
+
+
+CREATE TABLE `Review` (
+  
+`id` int (10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  
+`title` varchar(255) NOT NULL,
+  
+`description` text NOT NULL,
+  
+`meal_id` int(10) unsigned  NOT NULL,
+  
+`stars` int (5) unsigned  NULL,
+  
+`created_date` date
+ NOT NULL,
+);
+
+
+
+INSERT INTO Meal (Title, `Description`, Location, `When`, Max_reservations, Price, Created_date) 
+VALUES ('Meal001', 
+'With tomat', 
+'København H', 
+'2021-01-10 14:30:00', 
+'6',
+29, 
+'2020-01-08 00:00:00'
+); 
+INSERT INTO Meal (Title, `Description`, Location, `When`, Max_reservations, Price, Created_date) 
+VALUES ('Meal002', 
+'without tomat', 
+'København H', 
+'2021-01-10 14:30:00', 
+'6',
+29, 
+'2020-01-08 00:00:00'
+); 
+
+INSERT INTO Reservation (Number_of_guests, Meal_id, Created_date) 
+VALUES (
+"2",
+"2", 
+'2020-01-11'
+);
+INSERT INTO Reservation (Number_of_guests, Meal_id, Created_date) 
+VALUES (
+"5",
+"1", 
+'2020-01-10'
+);
+
+INSERT INTO Review (Title, `Description`, Meal_id, Stars, Created_date) 
+VALUES (
+'Excellent', 
+'Excellent food',
+'2', 
+4, 
+'2020-02-14'
+);INSERT INTO Review (Title, `Description`, Meal_id, Stars, Created_date) 
+VALUES (
+'Very good', 
+'Very good food',
+'1', 
+3, 
+'2020-02-14'
+);
+
+-- Get meals that has a price smaller than a specific price fx 90
+
+select * from meal where price < 90
+
+
+-- Get meals that still has available reservations...
+
+select Meal.Id,Meal.Title, Meal.max_reservation ,sum(Reservation.number_of_guests) as total
+from Meal
+inner join Reservation on Reservation.meal_id = Meal.Id
+having total < max_reservation;
+

```

Comment:
There are couple of problems here.

`max_reservation `does not exit, change it to `max_reservations` please

You have to use GROUP BY to check every meal individually

please try this query:
```

SELECT 
    Meal.Id,
    Meal.Title,
    Meal.max_reservations,
    SUM(Reservation.number_of_guests) AS total
FROM
    Meal
        INNER JOIN
    Reservation ON Reservation.meal_id = Meal.Id
**GROUP BY meal.id**
HAVING total < meal.max_reservations;
```